packages/aws_sdk/smoke_test/test/s3_test.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:aws_common/aws_common.dart';
+import 'package:aws_signature_v4/aws_signature_v4.dart';
+import 'package:smithy_aws/smithy_aws.dart';
+import 'package:smoke_test/src/sdk/s3.dart';
+import 'package:test/test.dart';
+
+import 'common.dart';
+
+void main() {
+  group('S3', () {
```

Comment:
should also add tests for APIs related to multipart upload?

Reply:
We can keep expanding it in future PRs. This was just to get a baseline.